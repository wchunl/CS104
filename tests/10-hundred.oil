fail:     .function
          .param ptr expr
          .param ptr file
          .param int line
          .end
putchr:   .function
          .param int chr
          .end
putint:   .function
          .param int num
          .end
putstr:   .function
          .param ptr str
          .end
getchr:   .function int
          .end
getstr:   .function
          .end
getln:    .function
          .end
main:     .function int
          .local int count
          count = 0
.wh0:     $t0:i = count < 16
          goto .od0 if not $t0:i
.do0:     $t1:i = count + 1
          count = $t1:i
          putint(count)
          putchr('\n')
.od0:     return 0
          return 
          .end
