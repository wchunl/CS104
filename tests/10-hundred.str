string_set[   0]:    4706206694200543916 0x24e5c08->","
                   16579916199511856337 0x24e7098->"getln"
string_set[   1]:    2297668033614959926 0x24e7818->"0"
string_set[   3]:   11868120863286193964 0x24e72b8->"main"
                    9611428142580219387 0x24e1608->"fail"
string_set[   4]:      21606623943142546 0x24e7508->"{"
                    4017852018407011111 0x24e5cf8->"file"
string_set[   6]:   13686721134187162920 0x24e8058->"}"
string_set[   8]:    1370730858159036685 0x24e7bc8->"+"
                    9386170080689500517 0x24e5ad8->"expr"
string_set[   9]:   10319632053171904911 0x24e7948->"while"
string_set[  12]:   15816506932648033945 0x24e5968->"("
string_set[  17]:   18413295556186398326 0x24e60f8->"putint"
string_set[  18]:    7211130878751977441 0x24e6228->"putchr"
                   12679825168387489523 0x24e6708->"num"
string_set[  23]:   11166701219533395393 0x24e6e58->"getstr"
                   12868445110721718657 0x24e5d18->")"
string_set[  28]:   14635384904416252381 0x24e78b8->"return"
string_set[  29]:   12291392672316406109 0x24e5ea8->"line"
string_set[  31]:    6192799629375248351 0x24e7eb8->"'\n'"
                    6142509188972423790 0x24e15e8->""
                   13348923631108115385 0x24e64e8->"putstr"
string_set[  32]:   11597697714117577063 0x24e7778->"="
string_set[  33]:    1583576014053361350 0x24e6888->"str"
string_set[  34]:    1782657156151550167 0x24e6038->"count"
string_set[  38]:   12597418120160561823 0x24e5b48->"int"
string_set[  39]:    7296690267450279874 0x24e6368->"chr"
string_set[  40]:    7282951971071689374 0x24e7b28->"16"
                   11911796782241914524 0x24e1218->"void"
string_set[  42]:    5344662657899890615 0x24e5ec8->";"
string_set[  44]:   13497578857207247412 0x24e5a08->"string"
string_set[  45]:   10159970873491820195 0x24e7e18->"1"
                    1396094175042499165 0x24e7a88->"<"
string_set[  46]:    9201800968376702500 0x24e6c38->"getchr"
load_factor = 0.723
bucket_count = 47
max_bucket_size = 3
