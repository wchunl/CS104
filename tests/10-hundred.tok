#  1 "10-hundred.oc"
#  1 "<built-in>"
#  1 "<command-line>"
#  1 "10-hundred.oc"
#  1 "oclib.h"
# 22 "oclib.h"
   6 22.00001 261 TOK_VOID      void
   6 22.00006 282 TOK_IDENT     fail
   6 22.00011  40 '('           (
   6 22.00012 263 TOK_STRING    string
   6 22.00019 282 TOK_IDENT     expr
   6 22.00023  44 ','           ,
   6 22.00025 263 TOK_STRING    string
   6 22.00032 282 TOK_IDENT     file
   6 22.00036  44 ','           ,
   6 22.00038 262 TOK_INT       int
   6 22.00042 282 TOK_IDENT     line
   6 22.00046  41 ')'           )
   6 22.00047  59 ';'           ;
   6 24.00001 261 TOK_VOID      void
   6 24.00006 282 TOK_IDENT     putchr
   6 24.00013  40 '('           (
   6 24.00014 262 TOK_INT       int
   6 24.00018 282 TOK_IDENT     chr
   6 24.00021  41 ')'           )
   6 24.00022  59 ';'           ;
   6 25.00001 261 TOK_VOID      void
   6 25.00006 282 TOK_IDENT     putint
   6 25.00013  40 '('           (
   6 25.00014 262 TOK_INT       int
   6 25.00018 282 TOK_IDENT     num
   6 25.00021  41 ')'           )
   6 25.00022  59 ';'           ;
   6 26.00001 261 TOK_VOID      void
   6 26.00006 282 TOK_IDENT     putstr
   6 26.00013  40 '('           (
   6 26.00014 263 TOK_STRING    string
   6 26.00021 282 TOK_IDENT     str
   6 26.00024  41 ')'           )
   6 26.00025  59 ';'           ;
   6 28.00001 262 TOK_INT       int
   6 28.00005 282 TOK_IDENT     getchr
   6 28.00011  40 '('           (
   6 28.00012  41 ')'           )
   6 28.00013  59 ';'           ;
   6 29.00001 263 TOK_STRING    string
   6 29.00008 282 TOK_IDENT     getstr
   6 29.00014  40 '('           (
   6 29.00015  41 ')'           )
   6 29.00016  59 ';'           ;
   6 30.00001 263 TOK_STRING    string
   6 30.00008 282 TOK_IDENT     getln
   6 30.00013  40 '('           (
   6 30.00014  41 ')'           )
   6 30.00015  59 ';'           ;
#  4 "10-hundred.oc"
   7 5.00001 262 TOK_INT       int
   7 5.00005 282 TOK_IDENT     main
   7 5.00009  40 '('           (
   7 5.00010  41 ')'           )
   7 5.00012 123 '{'           {
   7 6.00004 262 TOK_INT       int
   7 6.00008 282 TOK_IDENT     count
   7 6.00014  61 '='           =
   7 6.00016 283 TOK_INTCON    0
   7 6.00017  59 ';'           ;
   7 7.00004 267 TOK_WHILE     while
   7 7.00010  40 '('           (
   7 7.00011 282 TOK_IDENT     count
   7 7.00017  60 '<'           <
   7 7.00019 283 TOK_INTCON    16
   7 7.00021  41 ')'           )
   7 7.00023 123 '{'           {
   7 8.00007 282 TOK_IDENT     count
   7 8.00013  61 '='           =
   7 8.00015 282 TOK_IDENT     count
   7 8.00021  43 '+'           +
   7 8.00023 283 TOK_INTCON    1
   7 8.00024  59 ';'           ;
   7 9.00007 282 TOK_IDENT     putint
   7 9.00014  40 '('           (
   7 9.00015 282 TOK_IDENT     count
   7 9.00020  41 ')'           )
   7 9.00021  59 ';'           ;
   7 10.00007 282 TOK_IDENT     putchr
   7 10.00014  40 '('           (
   7 10.00015 284 TOK_CHARCON   '\n'
   7 10.00019  41 ')'           )
   7 10.00020  59 ';'           ;
   7 11.00004 125 '}'           }
   7 12.00004 268 TOK_RETURN    return
   7 12.00011 283 TOK_INTCON    0
   7 12.00012  59 ';'           ;
   7 13.00001 125 '}'           }
