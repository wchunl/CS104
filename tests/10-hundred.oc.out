; 0.  1: # 1 "10-hundred.oc"; 1.  0: 
; 1.  1: # 1 "<built-in>"
; 2.  1: # 1 "<command-line>"
; 3.  1: # 1 "10-hundred.oc"
; 4.  1: 
; 4.  2: 
; 4.  3: # 1 "oclib.h" 1
; 5.  1: # 22 "oclib.h"
; 6. 22: void fail (string expr, string file, int line);
; 6. 23: 
; 6. 24: void putchr (int chr);
; 6. 25: void putint (int num);
; 6. 26: void putstr (string str);
; 6. 27: 
; 6. 28: int getchr();
; 6. 29: string getstr();
; 6. 30: string getln();
; 6. 31: # 4 "10-hundred.oc" 2
; 7.  4: 
; 7.  5: int main() {
; 7.  6:    int count = 0;
; 7.  7:    while (count < 16) {
; 7.  8:       count = count + 1;
; 7.  9:       putint (count);
; 7. 10:       putchr ('\n');
; 7. 11:    }
; 7. 12:    return 0;
; 7. 13: }
hello